lib/request.js

```diff
@@ -319,7 +319,14 @@ module.exports = {
     var type = this.get('Content-Type');
     if (!type) return '';
 
-    return contentType.parse(type).parameters.charset || '';
+    try {
+      type = contentType.parse(type);
+    } catch (e) {
+      if (e instanceof TypeError) return '';
+      throw e
```

Comment:
Not sure I understood your comment. The error is specifically a TypeError - https://github.com/jshttp/content-type/blob/master/index.js#L88.